File: backend/routes/job_routes.py
from flask import Blueprint
from controllers.job_controller import post_job, list_jobs, get_job, edit_job, remove_job
from utils.auth import token_required, employer_required

job_bp = Blueprint('jobs', __name__)

job_bp.route('/', methods=['GET'])(list_jobs)
job_bp.route('/<int:job_id>', methods=['GET'])(get_job)

# Protected routes for employers
job_bp.route('/', methods=['POST'])(token_required(employer_required(post_job)))
job_bp.route('/<int:job_id>', methods=['PUT'])(token_required(employer_required(edit_job)))
job_bp.route('/<int:job_id>', methods=['DELETE'])(token_required(employer_required(remove_job)))

File: backend/routes/jobRoutes.js
const express = require('express');
const { postJob, listJobs, getJob, editJob, removeJob } = require('../controllers/jobController');
const { verifyToken, isEmployer } = require('../middleware/authMiddleware');

const router = express.Router();

router.get('/', listJobs);
router.get('/:job_id', getJob);

// Protected routes for employers
router.post('/', verifyToken, isEmployer, postJob);
router.put('/:job_id', verifyToken, isEmployer, editJob);
router.delete('/:job_id', verifyToken, isEmployer, removeJob);

module.exports = router;
